### PR TITLE
adding console command to migrate encryption keys - refs #13450

### PR DESCRIPTION
**Pull request by DeepDiver1975**
refs #13450

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@PVince81 @blizzz @schiesbn

**Comment by DeepDiver1975** [comment]
With this console command we can provide a workaround/fall back for key migration of external users and for high amounts of users

**Comment by DeepDiver1975** [comment]
```bash
deepdiver@alien:~/Development/ownCloud/core [encryption-key-migration-as-repair-step]$ ./occ encryption:migrate-keys
Reorganize system folder structure
Migrating keys for users on backend Database
   admin
Migrating keys for users on backend OCA\user_shibboleth\USER_SHIBBOLETH
   myself@testshib.org

**Comment by blizzz** [comment]
/me is testing with LDAP, please don't merge before

**Comment by blizzz** [comment]
It does not work well, yet, on non-default user folders, there is a mess afterwards:

```
# sudo -u www-data ./occ encryption:migrate-keys 8a685e8a-423b-1033-9e63-ad5f418ee02e
Reorganize system folder structure
Migrating keys   8a685e8a-423b-1033-9e63-ad5f418ee02e
# ls data/zombie4499/files_encryption/
8a685e8a-423b-1033-9e63-ad5f418ee02e.private.key  8a685e8a-423b-1033-9e63-ad5f418ee02e.privateKey  keyfiles  keys  share-keys
```

When adding setupFS() (as assumed) it seems to work well. I'll commit it.

File: apps/files_encryption/lib/migration.php [hunk 1]
```diff
@@ -60,27 +69,28 @@ public function reorganizeFolderStructure() {
...
-				$trashPath = $user . '/files_trashbin/keyfiles';
-				if (\OC_App::isEnabled('files_trashbin') && $this->view->is_dir($trashPath)) {
-					$this->renameFileKeys($user, $trashPath, true);
-					$this->view->deleteAll($trashPath);
-					$this->view->deleteAll($user . '/files_trashbin/share-keys');
-				}
-				// delete old folders
-				$this->deleteOldKeys($user);
+
+	public function reorganizeFolderStructureForUser($user) {
```
Comment (nickvergessen) [inline apps/files_encryption/lib/migration.php]: Does this have test coverage? *me hides*

Comment (DeepDiver1975) [inline apps/files_encryption/lib/migration.php]: yes - called here https://github.com/owncloud/core/pull/13604/files#diff-cee6b663fff8a76e0ca7b138c0f5904cR47

**Comment by DeepDiver1975** [comment]
@schiesbn can you please review this? THX

File: apps/files_encryption/lib/migration.php (hunk 1 above)
Comment (nickvergessen) [inline apps/files_encryption/lib/migration.php]: But it's missing a doc block (likewise the other methods in this class)

**Comment by schiesbn** [comment]
code looks good :+1: 

We should make sure to add this to the upgrade documentation to make sure that people know and use the command. cc @carlaschroder

**Comment by DeepDiver1975** [comment]
thanks for the review @schiesbn - I'm going to merge this then

**Comment by DeepDiver1975** [comment]
@carlaschroder shall we open a ticket on the doc repo? THX

**Comment by carlaschroder** [comment]
nope, I gottit @DeepDiver1975

**Pull request merged**